saleor/webhook/transport/asynchronous/transport.py

```diff
@@ -470,11 +479,13 @@ def trigger_webhooks_async(
 
 @app.task(bind=True)
 @allow_writer()
+@task_with_telemetry_context
 def generate_deferred_payloads(
     self,
     event_delivery_ids: list,
     deferred_payload_data: dict,
     send_webhook_queue: str | None = None,
+    span_links: Sequence[Link] | None = None,
```

Comment:
I have second thoughts about this bit where it looks like the task is expecting `span_links`, while we call it with `telemetry_context`, which gets rewritten in the decorator. Would it be possible to supply the global attributes inside the `get_current_context` function? Then I think whole decorator could be skipped and we could simply pass `telemetry_context` (as dict) between tasks.

Reply:
I agree with you, I changed it. But I propose to stick with the `task_with_telemetry_context` decorator, as apart from deserialization it also sets global attributes for the whole task. I fixed the decorator's docstring to describe it better. Is it ok for you?